src/napari_imagej/_function.py

```diff
@@ -233,21 +286,65 @@ def _napari_module_param_additions(module_info) -> Dict[str, Tuple[type, Any]]:
         additional_params["display_results_in_new_window"] = (bool, False)
     return additional_params
 
+def _is_non_default(input):
+    if not input.isRequired(): return False
+    if input.getDefaultValue() is not None: return False
+    return True
+
+
+def _sink_optional_inputs(inputs):
+    """
+    Python functions cannot have non-default args after a default arg.
+    We need to move all default inputs after the non-default ones.
+    """
+    sort_key = lambda x: -1 if _is_non_default(x) else 1
+    return sorted(inputs, key=sort_key)
+
+
+def _param_default_or_none(input):
+    """
+    Gets the Python function's default value, if it exists, for input.
+    """
+    default = input.getDefaultValue()
+    if default is not None:
+        try:
+            default = ij.py.from_java(default)
+        except Exception:
+            pass
+    return default
+
+
+def _param_annotation(input):
+    """
+    Gets the (Python) type hint for input
+    """
+    type = _ptype(input)
+    if not input.isRequired():
+        type = Optional[type]
+    return type
+
+
+def _module_param(input):
+    """Converts a java ModuleItem into a python Parameter"""
+    name = ij.py.from_java(input.getName())
+    kind = Parameter.POSITIONAL_OR_KEYWORD
+    default = _param_default_or_none(input)
+    annotation = _param_annotation(input)
+
+    if _is_non_default(input):
```

Comment:
I remember needing this if-else, but we should describe *why* we need it if we actually do need it.

Reply:
Described in 0a27a3f